text/0510-dynamodb-global-table.md

```diff
@@ -0,0 +1,1742 @@
+
+```
+arn:aws:dynamodb:us-east-1:123456789012:table/testddbstack-myDynamoDBTable-012A1SL7SMP5Q/stream/2015-11-30T20:10:00.000
+```
+
+Due to this, there is no way to reconstruct it for another region since we
+would not be aware of the
+timestamp value. If we want to support this, we can add a lookup function
+to determine the stream arn
+or create a CloudFormation custom resource.
+
+### Is this a breaking change?
+
+This is not a breaking change. This is adding functionality to CDK library.
+
+### What alternative solutions did you consider?
+
+#### Defaults for read and write capacity
+
+In the proposed solution, if a user selects billing mode as provisioned,
+then they would need to specify
+values for read and write capacities. There are no defaults assigned in
+this solution.
+
+In my opinion, if a user is using the provisioned mode, then they must
+make conscious decisions about what
+the capacity should look like for the table and its replicas.
+
+#### Kms key not created for customer managed keys
+
+In the proposed solution, if a user specifies customer managed key as
+their choice on encryption, then they
+would need to specify a key for each replica.
+
+Unlike the Table construct, I am not creating the KMS keys for the customers.
+I believe customer is making a conscious decision of using such encryption
+and should add relevant keys.
+
+To mitigate customer pain in this scenario, we can use a
+[multi region kms key](https://docs.aws.amazon.com/kms/latest/developerguide/multi-region-keys-overview.html)
+ and provision that if none is present.
+
+### What are the drawbacks of this solution?
+
+#### Code Redundancy and maintainance
+
+This solution uses a lot of code that is shared with the Table construct. Even if we try to maximize the
+code shared between Table and Global Table constructs, there would still be some repeated code in each of
+these. This could lead to added maintenance load in the long run since an update to the repeated code in
+one of construct probably would need to be reflected in other as well. If such an update is missed in one
+of the construct, it could lead to customer pain.
+
+### What is the high-level project plan?
+
+After this RFC is approved, I will start with implemetation and adding necessary tests.
+
+### Are there any open issues that need to be addressed later?
+
+#### Deprecating CloudFormation Custom Resource Global Table Support
+
+Since L2 support would now be added that is using the CloudFormation resource, there is no need of the
+custom resource solution. This should be deprecated and users must be informed of this change.
+
+#### RFC for migration path from Table construct to GlobalTable construct
+
+There would be potential users who would like to move from Table construct to GlobalTable construct in
+their CDK app. We need to come up with a solution to support this use case.
```

Comment:
This will be a blog post describing the steps to go through. At this point there's no need for design anymore, doesn't have to be an RFC.